crates/interfaces/src/provider.rs

```diff
@@ -122,6 +122,9 @@ pub enum ProviderError {
     /// Snapshot file is not found for requested transaction.
     #[error("not able to find {0} snapshot file for transaction id {1}")]
     MissingSnapshotTx(SnapshotSegment, TxNumber),
+    /// Error during conversion from U256 to u64.
+    #[error("failed to convert U256 to u64")]
+    ConversionError(U256),
```

Comment:
this is a bit too unspecific and too specific at the same time.

We want either `BlockNumberOverflow` or a general-purpose Overflow error.

I think the former would be more appropriate

Reply:
fixed